src/plugins/blocksmith/components/eme-form-block-modal/index.jsx

```diff
@@ -0,0 +1,141 @@
+import Message from '@gdbots/pbj/Message';
+import NodeRef from '@gdbots/schemas/gdbots/ncr/NodeRef';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {
+  Button,
+  FormGroup,
+  Input,
+  Label,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+} from '@triniti/admin-ui-plugin/components';
+
+import getEmeFormBlockRef from './getEmeFormBlockRef';
+
+export default class EmeFormBlockModal extends React.Component {
+  static propTypes = {
+    block: PropTypes.instanceOf(Message).isRequired,
+    isFreshBlock: PropTypes.bool.isRequired,
+    isOpen: PropTypes.bool,
+    toggle: PropTypes.func.isRequired,
+    onAddBlock: PropTypes.func.isRequired,
+    onEditBlock: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    isOpen: false,
+  };
+
+  constructor(props) {
+    super(props);
+
+    const { block } = props;
+
+    this.state = {
+      errorMsg: '',
+      touched: false,
+      isValid: block.has('form_ref'),
+      formRef: block.get('form_ref'),
+    };
+
+    this.handleAddBlock = this.handleAddBlock.bind(this);
+    this.handleChangeFormRef = this.handleChangeFormRef.bind(this);
+    this.handleEditBlock = this.handleEditBlock.bind(this);
+  }
+
+  componentDidMount() {
+    this.inputElement.focus();
+  }
+
+  handleAddBlock() {
+    const { onAddBlock, toggle } = this.props;
+    onAddBlock(this.setBlock());
+    toggle();
+  }
+
+  handleEditBlock() {
+    const { onEditBlock, toggle } = this.props;
+    onEditBlock(this.setBlock());
+    toggle();
+  }
+
+  handleChangeFormRef(event) {
+    const input = event.target.value;
+    const id = getEmeFormBlockRef(input);
```

Comment:
@albertgcsula `id` is an odd name for this variable eh? its null or a noderef string. and the util itself - why not just have an `isValidNodeRef` and use `NodeRef.fromString` inside?

Reply:
@efjacobson code updated.